docs/connections.md

```diff
@@ -493,6 +549,104 @@ TODO:
 
 TODO: `b:A` — does this add constraint to `a` or just `b` ? 
 
+# Advanced desugaring
+This section provides more examples of desugaring for various code constructs.
+This should give a better understanding of why the rules are as presented.
+
+Note that the desugaring translated below is very low-level.
+
+All assignments and code blocks are removed.
+
+
+## Out of order variable usage in block
+If within a block an identifier is used in the line before it is assigned, the
+`fix` function appropriate for the block's monadic context will be introduced.
```

Comment:
I don't know how performantly we can do this without a typechecker. The implementation doesn't currently support it, and I'll need to talk to Marcin about it when he's back.